pkg/p2p/streamtest/streamtest_test.go

```diff
@@ -314,6 +314,72 @@ func TestRecorder_closeAfterPartialWrite(t *testing.T) {
 			return err
 		}
 
+		// stream should be closed and write should return err
```

Comment:
the comment says `stream should be closed and write should return error`, but below we check if the error is not nil, then we wrap it, so i expect it to fail here and to return the error on line 319. but it doesn't, otherwise the test would fail in line 331. afaik `Close` should close the stream for _writing_ right? 
```go
// Stream is a bidirectional io pipe within a connection.
type MuxedStream interface {
	io.Reader
	io.Writer

	// Close closes the stream for writing. Reading will still work (that
	// is, the remote side can still write).
	io.Closer
        ...
```
so i would assume we should actually check in `318`:
`if _, err := rw.WriteString("expect err message"); err == nil {`
WDYT?

Reply:
This is an old test, so I did not want to change the approach. I think that result is the similar, but the error is returned on flush, since it is buffered. I think we would need to use a different reader writer implementation. Do you think we should change it?

Reply:
I can actually do this easily if I wrap existing write with a method that will do flush as well.

Reply:
yeah true, the original test checked only for read (which i believe is in fact incorrect). as you wish, we can keep this section as is and iterate :+1: